src/sentry/flags/endpoints/hooks.py

```diff
@@ -32,16 +26,13 @@ def post(self, request: Request, organization: Organization, provider: str) -> R
             return Response("Not enabled.", status=404)
 
         try:
-            if not validate_provider_event(
-                provider,
-                request.body,
-                request.headers,
-                organization.id,
-            ):
-                return Response("Not authorized.", status=401)
+            provider_cls = get_provider(organization.id, provider, request.headers)
```

Comment:
```suggestion
            provider_cls = get_provider(organization.id, provider, request.headers)
            if not provider_cls:
                return Response("Invalid provider.", status=400)
```
Can have get_provider return None instead of raising, so this is 400 instead of 500